src/HIE/Bios/Environment.hs

```diff
@@ -47,9 +47,20 @@ initSession  ComponentOptions {..} = do
         $ setVerbosity 0                       -- Set verbosity to zero just in case the user specified `-vx` in the options.
         $ setLinkerOptions df''                 -- Set `-fno-code` to avoid generating object files, unless we have to.
         )
+
+    let targets' = normaliseTargets componentRoot targets
     -- Unset the default log action to avoid output going to stdout.
     unsetLogAction
-    return targets
+    return targets'
+
+----------------------------------------------------------------
+
+normaliseTargets :: FilePath -> [G.Target] -> [G.Target]
+normaliseTargets wdir = map (\target -> target {G.targetId = normaliseTargetId wdir (G.targetId target)})
+
+normaliseTargetId :: FilePath -> G.TargetId -> G.TargetId
+normaliseTargetId wdir (G.TargetFile fp phase) = G.TargetFile (prependIfRelative wdir fp) phase
+normaliseTargetId _ tid = tid
```

Comment:
This should probably be named 'makeTargetsAbsolute' to mirror the naming of `makeDynFlagsAbsolute`.
@jneira, @bubba opinions?

Reply:
Yeah normalise to me means something more like `canonicalizePath`